spharpy/ambi.py

```diff
@@ -0,0 +1,274 @@
+            self,
+            data,
+            sampling_rate,
+            n_max,
+            basis_type,
+            normalization,
+            channel_convention,
+            condon_shortley,
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """Create Ambisonicssignal with data, and sampling rate.
+
+        Parameters
+        ----------
+        data : ndarray, double
+            Raw data of the signal in the time or frequency domain. The memory
+            layout of data is 'C'. E.g. data of ``shape = (3, 2, 1024)`` has
+            3 x 2 channels with 1024 samples or frequency bins each. Time data
+            is converted to ``float``. Frequency is converted to ``complex``
+            and must be provided as single sided spectra, i.e., for all
+            frequencies between 0 Hz and half the sampling rate.
+        sampling_rate : double
+            Sampling rate in Hz
+        n_max : int
+            Maximum spherical harmonic order
+        basis_type : str, optional
+            Type of spherical harmonic basis, either ``'complex'`` or
+            ``'real'``. The default is ``'complex'``.
+        normalization : str, optional
+            Normalization convention, either ``'n3d'``, ``'maxN'`` or
+            ``'sn3d'``. The default is ``'n3d'``.
+            (maxN is only supported up to 3rd order)
+        channel_convention : str, optional
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        condon_shortley : bool, optional
+            Whether to include the Condon-Shortley phase term.
+            The default is True.
+        n_samples : int, optional
+            Number of samples of the time signal. Required if domain is
+            ``'freq'``. The default is ``None``, which assumes an even number
+            of samples if the data is provided in the frequency domain.
+        domain : ``'time'``, ``'freq'``, optional
+            Domain of data. The default is ``'time'``
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+            or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+            for more information. The default is ``'none'``, which is typically
+            used for energy signals, such as impulse responses.
+        comment : str
+            A comment related to `data`. The default is ``None``.
+
+        References
+        ----------
+        ..
+
+        """
+        # TODO: discuss if ambisonics channels always on first dimension?
+        if not data.shape[1] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few coefficients '
+                             'for N = {n_max}.')
```

Comment:
I think it would make sense to go for shape[-2], as for example band filter functions add extra leading dimensions, which would require reshaping signals to keep the "SH" dimension as leading dimension.